### PR TITLE
added potential evaporation to offline output, changed checks range, …

### LINKED ISSUE #335: facilitate output of potential evaporation directly from the offline code base
There are many applications where direct knowledge of potential evaporation (and/or its relation to simulated evaporation or evapotranspiration) would be of interest. However this is not facilitated by CABLE despite the `canopy%epot` variable carrying this information.  While an approximation of average potential evaporation could be estimated from other outputs, the nonlinearity in the equations implies that it would be better evaluated by CABLE's inbuilt time averaging.

Note that `ssnow%potev` in CABLE is more strictly the 'latent heat flux consistent with potential evaporation from the soil' not evaporation itself - i.e. units are currently W/m2 not mm of water/s - this may need to be handled within this issue.

### PR DESCRIPTION
**Pull request by har917**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

This PR adds the potential evaporation diagnostic (`canopy%epot`) to fluxes section of the cable output. 
A minor correction to the diagnostic has also been applied - which could lead to impacts in the coupled model (via `%wetfac_cs` and JULES variable `resft`) - and the checks_ranges updated.

Fixes #335

## Type of change

Please delete options that are not relevant.

- [x] Bug fix
- [x] additional output

## Checklist

- [x] The new content is accessible and located in the appropriate section.
- [ ] I have checked that links are valid and point to the intended content.
- [ ] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

----
📚 Documentation preview 📚: https://cable--346.org.readthedocs.build/en/346/

### CODE REVIEW

**Comment by har917** [comment]
@ccarouge Holding off on benchcab testing for now given that MAIN is currently broken.

@AlisonBennett This changeset works for single site runs - and I see no reason why it shouldn't work in other situations.  We should be able `git cherry-pick` this across to BLAZE_9184.

**Comment by AlisonBennett** [comment]
@har917 can I cherry pick this one across to BLAZE_9184 now?  I will test it by default when doing the BLAZE 1000pt runs.

**Comment by har917** [comment]
> @har917 can I cherry pick this one across to BLAZE_9184 now? I will test it by default when doing the BLAZE 1000pt runs.

yes - should be ready

**Comment by AlisonBennett** [comment]
Probably the simplest way forward if we want Pot_Evap in the output :)

**Comment by har917** [comment]
0a69346fc8ed8cda575ce612481bf3929ddea2a3 reverts the evaluation of `%epot` to its original value - this allows `%wetfac_cs` to be as previous in connections to ACCESS.

Instead a revised evaluation of `%epot` has been included at line 671 of `cable_canopy` - scientifically we do not need to weight the vegetation and soil components of potential evaporation by `%transd`.  This weighting has already been included via the net radiation used to determine the component parts.

**Comment by har917** [comment]
testing of equivalent changes in the BLAZE_9184 branch - see 1f666d332bc6fcd7b61939233ee3b12af58c3075 and 0f297f2f5211c54b2bd141b1974eeb150251da4b - indicate that this extension leads to bitwise reproducible results on all other variables
Fluxes__Blaze_9184_PotEvap_Diff_acttest9_20240717_164348
Pools_tot__Blaze_9184_PotEvap_Diff_acttest9_20240717_164208
Evap_vs_PotEvap_Blaze_9184_PotEvap_acttest9_20240717_164951

**Comment by har917** [comment]
benchcab regression testing log here 
[benchmark_cable_qsub.sh.o121035128.txt](https://github.com/user-attachments/files/16277198/benchmark_cable_qsub.sh.o121035128.txt)

This (of course) fails because of the new output variable `PotEvap`

Somewhat more worryingly - all time varying outputs have also been impacted - see 
[BR-Sa3_2001-2003_FLUXNET2015_Met_S1_R0_R1_editted.txt](https://github.com/user-attachments/files/16277417/BR-Sa3_2001-2003_FLUXNET2015_Met_S1_R0_R1_editted.txt) for a selection from one site (the full diffs are too large).  All these differences are at the numerical precision level - and do not accumulate through time. 

Given the equivalent changes applied to the BLAZE_9184 branch produce bitwise equivalence on the other variables I think this has to be something to do with how the new output routine operate.

Sending to review for another opinion as to whether this merits further investigation.

**Comment by ccarouge** [comment]
@har917 Looking into it to see where the differences might be coming from so we can exclude the changes here.

**Review by ccarouge — APPROVED** [review]
I've looked into the numerical precision differences. It turns out this is linked to the fact we load openmpi in benchcab even when we do serial compilations. We'll need to look into that more in details in benchcab, but it means it's not a problem for this PR to go ahead.

I am approving but I have a question about where the update to `epot` calculation is placed.

File: src/science/canopy/cable_canopy.F90
Comment (har917) [reply src/science/canopy/cable_canopy.F90:673]: Ideally we would change `%epot` and hence `%wetfac_cs` in the section above this change - and use the correct value in both offline and coupled.  However since `%wetfac_cs` is one of the coupling variables - and one that I think does get used - I think this needs independent testing within ACCESS (and so is a separate issue).

In the interim I opted to adjust `%epot` after the evaluation of `%wetfac_cs` which will allow updates to CABLE3 to be ported across to ACCESS as a first step towards resyncing the code bases.

**Comment by SeanBryan51** [comment]
Restoring this branch to reproduce https://github.com/CABLE-LSM/benchcab/issues/304.

**Pull request merged**
